### PR TITLE
NPC breadcrumbs revived and expanded

### PR DESCRIPTION
**Pull request by LyleSY**
#### Summary
Content "NPC breadcrumbs revived and expanded"

#### Purpose of change

Mapgen can make it very difficult to find faction camps and static NPCs. Giving random spawning NPCs some knowledge of the world would help make things feel more connected, and improve player ability to find story quests and more.

#### Describe the solution

This adds a new random background story category for NPCs, "other survivors stories" or BOSS. These stories are randomly assigned on NPC creation just like BGSS stories are. Many of them will have a chance to include a mission leading to various interesting static NPCs and locations, or to their reanimated remains. Revives and expands #49375

Need to :

- [x] Figure out conflicts
- [x] New breadcrumbs for farmer
- [x] New breadcrumbs for cook
- [x] New breadcrumbs for graveyard lurker
- [x] Missions

#### Describe alternatives you've considered

A mutant fish NPC in the underwater lab with an underwater questline and fishy and athletics training rewards

#### Testing

Game runs no errors. Haven't managed to get it to fire in testing, not sure if more backend work is needed.

#### Additional context

Part of the great and mighty Abandoned but Desired PRs project. Original PR by @I-am-Erk

### CODE REVIEW

**Comment by I-am-Erk** [comment]
Appreciate it thanks. Let me know if I can help. The setup for the finding missions should be similar to the teamster character in the refugee center.

**Comment by LyleSY** [comment]
cardio test failure doesn't look related

File: data/json/npcs/BG_traits.json [hunk 1]
```diff
@@ -688,5 +692,58 @@
     "purifiable": false,
     "types": [ "BACKGROUND_SURVIVAL_STORY" ],
     "flags": [ "BG_SURVIVAL_STORY" ]
+  },
+  {
+    "type": "mutation",
+    "//": "This is a unique background story trait that should not be given out randomly.",
```
Comment (Maleclypse) [inline data/json/npcs/BG_traits.json:698]: Are any of these unique since I think they are all being distributed randomly above in the trait_group?

**Review by Maleclypse — COMMENTED** [review]
This looks great!

File: data/json/npcs/BG_traits.json (hunk 1 above)
Comment (LyleSY) [reply data/json/npcs/BG_traits.json:698]: Nope, just copied it over, good call

**Comment by Inglonias** [comment]
I love this. This is wonderful

**Comment by Brambor** [comment]
> Haven't managed to get it to fire in testing, not sure if more backend work is needed.

Makes sense, I don't believe it is in the game as it shares
```
    "id": "TALK_FRIEND_CONVERSATION",
    "type": "talk_topic",
```
with existing JSON entries.

Stumbled upon this while deleting duplicates in #69129

**Pull request merged**